pandas/core/frame.py

```diff
@@ -1399,11 +1399,44 @@ def to_dict(self, orient="dict", into=dict):
             )
         # GH16122
         into_c = com.standardize_mapping(into)
-        if orient.lower().startswith("d"):
+        orient = orient.lower()
+        # GH32515
+        if orient.startswith(("d", "l", "s", "r", "i")) and orient not in {
```

Comment:
you can just do orient not in (.....)

Reply:
but then the warning will raise even if the user uses an option which is not in the list.

Reply:
k this looks better